app/controlplane/pkg/biz/workflowcontract.go

```diff
@@ -601,22 +644,40 @@ func (uc *WorkflowContractUseCase) findProvider(providerName string) (*policies.
 
 // UnmarshalAndValidateRawContract Takes the raw contract + format and will unmarshal the contract and validate it
 func UnmarshalAndValidateRawContract(raw []byte, format unmarshal.RawFormat) (*Contract, error) {
-	contract := &schemav1.CraftingSchema{}
-	err := unmarshal.FromRaw(raw, format, contract, true)
-	if err != nil {
-		return nil, NewErrValidation(err)
-	}
-
-	// Custom Validations
-	if err := contract.ValidateUniqueMaterialName(); err != nil {
-		return nil, NewErrValidation(err)
-	}
-
-	if err := contract.ValidatePolicyAttachments(); err != nil {
-		return nil, NewErrValidation(err)
+	// Try parsing as v2 Contract format first
+	v2Contract := &schemav1.CraftingSchemaV2{}
+	v2Err := unmarshal.FromRaw(raw, format, v2Contract, true)
+	if v2Err == nil {
+		// Custom Validations
+		if err := v2Contract.ValidateUniqueMaterialName(); err != nil {
+			return nil, NewErrValidation(fmt.Errorf("unique material name validation failed: %w", err))
+		}
+		if err := v2Contract.ValidatePolicyAttachments(); err != nil {
+			return nil, NewErrValidation(fmt.Errorf("policy attachment validation failed: %w", err))
+		}
+		// Convert to v1 for backward compatibility with old CLIs
+		v1Schema := v2Contract.ToV1()
+		return &Contract{Raw: raw, Format: format, Schema: v1Schema, Schemav2: v2Contract}, nil
+	}
+
+	// Fallback to v1 CraftingSchema format
+	// DEPRECATED: v1 schema is deprecated, use v2 Contract format instead
+	v1Contract := &schemav1.CraftingSchema{}
+	v1Err := unmarshal.FromRaw(raw, format, v1Contract, true)
+	if v1Err == nil {
+		// Custom Validations
+		if err := v1Contract.ValidateUniqueMaterialName(); err != nil {
+			return nil, NewErrValidation(fmt.Errorf("unique material name validation failed: %w", err))
+		}
+		if err := v1Contract.ValidatePolicyAttachments(); err != nil {
+			return nil, NewErrValidation(fmt.Errorf("policy attachment validation failed: %w", err))
+		}
+		return &Contract{Raw: raw, Format: format, Schema: v1Contract}, nil
 	}
 
-	return &Contract{Raw: raw, Format: format, Schema: contract}, nil
+	// Both parsing attempts failed
+	// Best effort: provide errors for both schemas
+	return nil, NewErrValidation(fmt.Errorf("contract validation failed:\n  v2 Contract format error: %w\n  v1 CraftingSchema format error: %w", v2Err, v1Err))
```

Comment:
I think fmt.Errorf only allows 1 wrapped error, so in practice this will only take the later. Instead, use `multierror`, there are some usages already in the code.